src/bootstrap/test.rs

```diff
@@ -144,6 +144,45 @@ You can skip linkcheck with --exclude src/tools/linkchecker"
     }
 }
 
+#[derive(Debug, Copy, Clone, PartialEq, Eq, Hash)]
+pub struct HtmlCheck {
+    target: TargetSelection,
+}
+
+impl Step for HtmlCheck {
+    type Output = ();
+    const DEFAULT: bool = true;
+    const ONLY_HOSTS: bool = true;
+
+    fn should_run(run: ShouldRun<'_>) -> ShouldRun<'_> {
+        run.path("src/tools/html-checker")
+    }
+
+    fn make_run(run: RunConfig<'_>) {
+        run.builder.ensure(HtmlCheck { target: run.target });
+    }
+
+    fn run(self, builder: &Builder<'_>) {
+        if !Command::new("tidy")
+            .arg("--version")
+            .stdout(Stdio::null())
+            .status()
+            .map_or(false, |status| status.success())
+        {
+            eprintln!("not running HTML-check tool because `tidy` is missing");
+            return;
+        }
```

Comment:
Can you check if this is running in CI and give a hard error instead? You can do that with `crate::CiEnv::current() != crate::CiEnv::None`.

Reply:
Yup, this was skipped in the PR run:
```
Set({"src/tools/html-checker"}) not skipped for "bootstrap::test::HtmlCheck" -- not in ["src/tools/tidy"]
not running HTML-check tool because `tidy` is missing
```

Reply:
Yes since the command is not installed. I'll do something something similar to my eslint check that I added a while ago.

Reply:
What's the point of a check that only works locally? We should test this in CI or it will constantly regress.

Reply:
We only need it to work on at least one CI, no need to run it everywhere.

Reply:
You're not testing if it runs on at least one CI. If you forget to install tidy, CI will silently pass.

Reply:
I think this is likely going to want the same resolution we're (coming to) on the browser-ui-test thread, which is that we don't run it by default unless tidy is available (e.g., this check passes). We'd still run it if tidy is available, and if it is explicitly requested.

In the future we'll want to develop a solution that works better for these 'opt-in' style tests, but this should be OK for now.